src/ORM/Connect/MySQLiConnector.php

```diff
@@ -63,7 +64,12 @@ public function prepareStatement($sql, &$success)
         // Record last statement for error reporting
         $statement = $this->dbConn->stmt_init();
         $this->setLastStatement($statement);
-        $success = $statement->prepare($sql);
+        try {
+            $success = $statement->prepare($sql);
+        } catch (mysqli_sql_exception $e) {
+            $success = false;
+            $this->databaseError($e->getMessage(), E_USER_ERROR, $sql);
```

Comment:
```suggestion
```
Sorry, I wasn't clear in my previous comment. I think we should either set `$success` to false _or_ throw the error, and I lean more towards setting `$success` to false _instead_ of throwing the error here since it preserves the previous behaviour.

In most cases like you say it won't matter - if this is hit through other methods (e.g. `preparedQuery`) then the database error is thrown is `$success` is false. But because this method is public, and we're going out of our way to catch the `mysqli_sql_exception`, I think we should give developers a consistent experience regardless of which php version they're running.

Reply:
I don't think we should be essentially suppressing a thrown `mysqli_sql_exception` exception here, it's generally better to have hard fails then silent fails so that it's easier to see what's going on.  That seems like it was the rationale for PHP 8.1 to start throwing exceptions in the first place.  I don't think we should be working against that, I think it's a good change.

https://php.watch/versions/8.1/mysqli-error-mode does show how to either configure mysqli to throw exception in all versions of php - ` mysqli_report(MYSQLI_REPORT_ERROR|MYSQLI_REPORT_STRICT);` or suppress in all versions of php `mysqli_report(MYSQLI_REPORT_OFF);`.

I'm not sure how you'd actually set this configure in the context of Silverstripe, though I'm sure there will be some way to do this.  If someone was to set config to always throw exceptions in all versions of php, this if we surpress via 
```
} catch (mysqli_sql_exception $e) {
    $success = false;
}
```
Then it seems like we're overriding their config for how they want mysqli to behave.

Reply:
Maybe add a `@throws` PHP doc statement.

Reply:
I don't think we need to add the `@throws` here, since it's already documented in the `databseError()` method - https://github.com/silverstripe/silverstripe-framework/blob/7a88f91c13cc1e004604dcccf34d5adb1271ee6b/src/ORM/Connect/DBConnector.php#L47